oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/shardingsphere/RangeDatabaseShardingAlgorithm.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.shardingsphere;
+
+import org.apache.shardingsphere.api.sharding.standard.RangeShardingAlgorithm;
+import org.apache.shardingsphere.api.sharding.standard.RangeShardingValue;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.LinkedHashSet;
+
+/**
+ * Range sharding algorithm to shard database.
+ *
+ * Sharding Sphere will use it to handle SQL with between query condition.
+ */
+public final class RangeDatabaseShardingAlgorithm implements RangeShardingAlgorithm<String> {
```

Comment:
What is this for? And what is the difference between this and `PreciseDatabaseShardingAlgorithm`?

Reply:
Please note the small difference between `handle SQL with between(>= / <=) query condition` and `handle SQL with equal or in (= / IN) query condition`.
Sorry my comments confused you, do you have a idea to describe them better?